src/destreamer.ts

```diff
@@ -208,70 +143,92 @@ async function downloadVideo(videoUrls: string[], outputDirectory: string, sessi
         return;
     }
 
-    await Promise.all(metadata.map(async video => {
-        console.log(colors.blue(`\nDownloading Video: ${video.title}\n`));
+    const outDirsIdxInc = outputDirectories.length > 1 ? 1:0;
+    for (let i=0, j=0, l=metadata.length; i<l; ++i, j+=outDirsIdxInc) {
+        const video = metadata[i];
+        let previousChunks = 0;
+        const pbar = new cliProgress.SingleBar({
+            barCompleteChar: '\u2588',
+            barIncompleteChar: '\u2591',
+            format: 'progress [{bar}] {percentage}% {speed} {eta_formatted}',
+            barsize: Math.floor(process.stdout.columns / 3),
+            stopOnComplete: true,
+            hideCursor: true,
+        });
 
-        video.title = makeUniqueTitle(sanitize(video.title) + ' - ' + video.date, argv.outputDirectory);
+        console.log(colors.yellow(`\nDownloading Video: ${video.title}\n`));
 
-        const outputPath = outputDirectory + path.sep + video.title + '.mp4';
+        video.title = makeUniqueTitle(sanitize(video.title) + ' - ' + video.date, outputDirectories[j]);
 
         // Very experimental inline thumbnail rendering
         if (!argv.noThumbnails)
             await drawThumbnail(video.posterImage, session.AccessToken);
 
         console.info('Spawning ffmpeg with access token and HLS URL. This may take a few seconds...\n');
 
-        ffmpeg()
-            .input(video.playbackUrl)
-            .inputOption([
-                // Never remove those "useless" escapes or ffmpeg will not
-                // pick up the header correctly
-                // eslint-disable-next-line no-useless-escape
-                '-headers', `Authorization:\ Bearer\ ${session.AccessToken}`,
-            ])
-            .format('mp4')
-            .saveToFile(outputPath)
-            .on('codecData', data => {
-                console.log(`Input is ${data.video} with ${data.audio} audio.\n`);
-
-                pbar.start(video.duration, 0, {
-                    speed: '0'
-                });
-
-                process.on('SIGINT', () => {
-                    pbar.stop();
-                });
-            })
-            .on('progress', progress => {
-                const currentChunks = ffmpegTimemarkToChunk(progress.timemark);
-
-                pbar.update(currentChunks, {
-                    speed: progress.currentKbps
-                });
-            })
-            .on('error', err => {
-                pbar.stop();
-                console.log(`ffmpeg returned an error: ${err.message}`);
-            })
-            .on('end', () => {
+        const outputPath = outputDirectories[j] + path.sep + video.title + '.mp4';
+        const ffmpegInpt = new FFmpegInput(video.playbackUrl, new Map([
+            ['headers', `Authorization:\ Bearer\ ${session.AccessToken}`]
+        ]));
+        const ffmpegOutput = new FFmpegOutput(outputPath);
+        const ffmpegCmd = new FFmpegCommand();
+
+        pbar.start(video.totalChunks, 0, {
```

Comment:
Placing this comment here however not sure if this is what causes it - 
![p](https://user-images.githubusercontent.com/6472374/79205527-ba804380-7e46-11ea-8ccd-6fc9d0ffa065.gif)

See how the progress bar shows up before the download actually starts or ffmpeg is invoked?
Can we make it start on the first chunk? The user experience will slightly improve. We could also print an extra message before the progress bar shows up: `Fetching key to decrypt stream...`.

Reply:
yeah, this is not bug, i made it like so because i thought it was ugly, you usually see the progress bar way before a download starts, and now switching to fessonia we don't have a start event, so it's the only way anyway

Reply:
No worries, let's keep it like it is.